pkg/api/v1/types.go

```diff
@@ -770,8 +770,16 @@ type SecretVolumeSource struct {
 	// the volume setup will error. Paths must be relative and may not contain
 	// the '..' path or start with '..'.
 	Items []KeyToPath `json:"items,omitempty" protobuf:"bytes,2,rep,name=items"`
+	// Mode bits to use on created files by default. The final mode bits
+	// will be the provided value AND 0777.
+	// Directories within the path are not affected by this setting
+	DefaultMode *int32 `json:"defaultMode,omitempty" protobuf:"bytes,3,opt,name=defaultMode"`
 }
 
+const (
+	SecretVolumeSourceDefaultMode int32 = 0644
```

Comment:
I think this needs to be `0755`, because the execute bit is needed to recurse directory trees if you don't own them.